Update Getting Started with services (#2257)
pivotal/docs-tap: getting-started/consume-services.hbs.md

# Consume services on Tanzu Application Platform

This tutorial walks the application developer through deploying two application workloads and configuring them to communicate via a service instance. It uses RabbitMQ as an example, but the process is the same regardless of the service you want to set up. You will use the `tanzu service` CLI plug-in and will learn about classes, claims and bindings.

## <a id="you-will"></a>What you will do

- Discover the range of services available to you
- Create a claim for an instance of one of the services
- Create two application workloads and bind them to the claim so that the workloads utilize the service instance

## <a id="overview"></a>Overview

The following diagram depicts a summary of what this tutorial covers.

![Diagram shows the default namespace and a glimpse at what is happening behind the scenes. The default namespace has two application workloads, each connected to a service binding. The service bindings refer to a single claim, which refers to a service instance.](../images/getting-started-stk-1.png)

Bear the following observations in mind as you work through this guide:

> 1. There are a set of four classes preinstalled on the cluster.

Reviewer: **thought:** Unless they have been explicitly disabled by the Service/Platform Operator.

Author: I had the same thought, but I think given this is the "getting started" section of the docs it's probably ok to assume that they haven't been disabled.

2. There is no configuration or setup required by Service Operators.

> 3. ProvisionedService is the contract allowing credentials and connectivity information to flow from the service instance to the resource claim, to the class claim, to the service binding, and ultimately to the application workload. For more information, see [ProvisionedService](https://github.com/servicebinding/spec#provisioned-service) on GitHub.

Reviewer: **thought:** I wonder whether this is actually essential for the audience of this particular getting started guide, or whether it potentially adds more confusion as there really isn't any API with the name `ProvisionedService` and the diagram linked above doesn't mention it either.

> 4. The life cycles of service bindings are implicitly tied to the life cycle of workloads.

Reviewer: ```suggestion
4. The life cycle of a service binding is implicitly tied to the life cycle of a workload, and hence managed by the Application Developer.
```

1. Have set up the `default` namespace to use installed packages and use it as your developer namespace.
For more information, see [Set up developer namespaces to use installed packages](../set-up-namespaces.md).
1. Ensure that your Tanzu Application Platform cluster can pull source code from GitHub.

## <a id="stk-discover"></a> Discover available services

This section covers:

> * Using `tanzu service class list` and `tanzu service class get` to find information about the classes.

Reviewer: **thought:** _"... the classes of services."_

Reviewer: ```suggestion
* Using `tanzu service class list` and `tanzu service class get` to find information about the classes of services.
```

You can create a claim for an instance of a class using the `tanzu service class-claim create` command. Let's create a claim for the `rabbitmq-unmanaged` class. We'll pass a parameter to the command to set the storage capacity of the resulting instance to 3 Gigabytes, rather than using the default 1 Gigabyte.

    ```console
    tanzu service class-claim create rabbitmq-1 --class rabbitmq-unmanaged --parameter storageGB=3
    ```

    Expected output:

    ```console
      Creating claim 'rabbitmq-1' in namespace 'default'.
    ```

You can get detailed information about the claim using the `tanzu service class-claim get` command.

    ```console
    tanzu service class-claim get rabbitmq-1
    ```

    Expected output:

    ```console
      Name: rabbitmq-1
      Namespace: default
      Claim Reference: services.apps.tanzu.vmware.com/v1alpha1:ClassClaim:rabbitmq-1
      Class Reference:
        Name: rabbitmq-unmanaged
      Parameters:
        storageGB: 3
      Status:
        Ready: True
        Claimed Resource:
          Name: b5982046-a1e9-40cf-8282-00fe67a2f868
          Namespace: default
          Group:
          Version: v1
          Kind: Secret
    ```

> It may take a moment or two for the claim to report `Ready: True`. Behind the scenes, the creation of the claim has triggered the on-demand creation of a helm release of the Bitnami RabbitMQ helm chart. Credentials and connectivity information required to connect to the RabbitMQ cluster have been formatted according to the [Service Binding Specification for Kubernetes](https://github.com/servicebinding/spec) and stored in a `Secret` in your namespace.

Reviewer: ```suggestion
It may take a moment or two for the claim to report `Ready: True`. Behind the scenes, the creation of the claim has triggered the on-demand creation of a Helm release of the Bitnami RabbitMQ Helm chart. Credentials and connectivity information required to connect to the RabbitMQ cluster have been formatted according to the [Service Binding Specification for Kubernetes](https://github.com/servicebinding/spec) and stored in a `Secret` in your namespace.
```

> However, as an application developer you don't need to worry about what's going on behind the scenes. Tanzu Application Platform promotes a strong separation of concerns between Service Operators, who are responsible for managing service instances for the platform, and Application Developers, who just want to use those service instances with their application workloads. The class and claims abstractions work in service of that separation of concerns. Application Operators/Developers create claims and Service Operators do whatever work is necessary to fulfil them.

Reviewer: ```suggestion
However, as an Application Developer you don't need to worry about what's going on behind the scenes. Tanzu Application Platform promotes a strong separation of concerns between Service Operators, who are responsible for managing service instances for the platform, and Application Developers, who just want to use those service instances with their application workloads. The class and claims abstractions work in service of that separation of concerns. Application Operators/Developers create claims and Service Operators do whatever work is necessary to fulfil them.
```
